x-pack/plugin/otel-data/src/main/resources/component-templates/traces-otel@mappings.yaml

```diff
@@ -0,0 +1,65 @@
+---
+version: ${xpack.oteldata.template.version}
+_meta:
+  description: Default mappings for OpenTelemetry traces
+  managed: true
+template:
+  mappings:
+    date_detection: false
```

Comment:
We definitely don't need it here 👍 

> And in traces@mappings for that matter, so maybe it doesn't need to be in `otel@mappings` either

That'd affect metrics from what I see. If we remove it from `otel@mappings`, then logs will be still ok, because `logs@mappings` has this as well, but from what I see, for metrics it's not set elsewhere. 

So removed it from here, but kept in `otel@mappings`.